### PR TITLE
[WNMGDS-808] Add `getA11yStatusMessage` to Autocomplete props

### PR DESCRIPTION
**Pull request by nichia**
## Summary
Specifically call out the `getA11yStatusMessage` downshift prop in React prop documentation

### CODE REVIEW

File: packages/design-system-docs/src/pages/components/Autocomplete/_Autocomplete.docs.scss [hunk 1]
```diff
@@ -44,6 +44,7 @@ Style guide: components.autocomplete.react
 - The `<Autocomplete>` component has taken special care to ensure accessibility for screenreader devices. It announces the number of results based on `items` matches with the `inputValue` string. The component also reads out the name of each list item when users arrow up or down.
 - `<Autocomplete>` allows developers to add hint text in the `<label>` element. This hint `<span>` is added to the markup by passing a  `String` into the `hint` prop.
 - `<Autocomplete>` has a button (styled visually as a link) to clear the search, and refocus the `<input>` element. This resets the local state `selectedItem` to `null`, and will re-read the label and screenreader hint text.
+- To customize the ARIA statuses when a prop is changed, access Downshift's [getA11yStatusMessage](https://github.com/downshift-js/downshift#geta11ystatusmessage) method and override the default message. This is made possible by the `<Autocomplete>` component which allows developers to pass extra props to access the underlying Downshift's component.
```
Comment (bernardwang) [inline packages/design-system-docs/src/pages/components/Autocomplete/_Autocomplete.docs.scss:47]: Can we actually add this to the Autocomplete props? You can look at `onInputValueChange` as a reference for how they document it

Comment (nichia) [reply packages/design-system-docs/src/pages/components/Autocomplete/_Autocomplete.docs.scss]: Thanks, have updated to add getA11yStatusMessage to Autocomplete props instead.

**Review by line47 — APPROVED** [review]

File: packages/design-system/src/types/Autocomplete/Autocomplete.d.ts [hunk 2]
```diff
@@ -53,6 +53,10 @@ export interface AutocompleteProps extends Omit<DownshiftProps<any>, PropsNotPas
    * attribute on a label and the id of an input.
    */
   id?: string;
+  /**
+   * Customize the default status messages announced to screenreader users via aria-live when autocomplete results are populated. [Read more on downshift docs.](https://github.com/paypal/downshift#geta11ystatusmessage)
+  */
+  getA11yStatusMessage?: (...args: any[]) => any;
```
Comment (bernardwang) [inline packages/design-system/src/types/Autocomplete/Autocomplete.d.ts:59]: Can we use downshift props here?
 i.e. `DownshiftProps<any>["onInputValueChange"];`
 
 The same can be done for other props like `itemToString`,

Comment (nichia) [reply packages/design-system/src/types/Autocomplete/Autocomplete.d.ts]: Changed the DownshiftProps typing for `DownshiftProps` `getA11yStatusMessage`  `itemToString`

**Review by bernardwang — APPROVED** [review]
nice work @nichia !

**Pull request merged**